spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,331 @@
+
+- Basically everything above is extremely ARM specific. IDs are represented in a different way for data planes and non-ARM endpoints, 
+  and those endpoints JSON schemas may not have the magical `resources` property which allows us to identify owner-dependent relationships.
+  How do we structure the code to solve our ARM problems but allow for alternate solutions for non-ARM?
+- Is subscription a parameter to in YAMLs, or is the operator configured to act only on a certain subscription?
+  If we only operate on a single sub do we not support cross-sub? Do we care?
+- ARM templates have `dependsOn` for specifying creating order of related resources (where A refers to B) - we don't have that, what does that mean?
+- What to do with awkward resources where the owner requires at least 1 dependent to also be created with it?
+  David Justice pointed out [this one](https://github.com/Azure/k8s-infra/blob/14105b1cb3f6967cd086c9f8f75fb16bb85d6318/apis/microsoft.network/v1/networkinterface_types.go#L43)
+- Do we want to use the same type for ownership relationships and "related" relationships? Ownership has other angles such
+  as how deletes propagate which in theory don't apply for other kinds of relationships.
+- Do we need to worry about letting customers choose between [foreground cascading deletion](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion)
+  and [background cascading deletion](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#background-cascading-deletion) or do we just pick 
+  one behavior which is best for our case?
+
+# The road not travelled
+## How to represent references
+### Use fully qualified ARM ID (a single string) for all references
+#### Pros
+- Super simple to implement, because it's what ARM expects at the end of the day anyway.
+
+#### Cons
+- You can't easily transplant your YAML between subscriptions/resource groups because those IDs are in the YAML - you need
+  templates and variables so that you can easily move between different resource groups or Subscriptions.
+- Customers can't stay in Kubernetes-land, they have to move their mental model to an "Azure" model.
+
+### Use k8s-infra style KnownTypeReference for all references
+#### Pros
+- References look very Kubernetes native.
+
+#### Cons
+- This requires that we detect and label all references appropriately.
+- It means that you can only reference other resources which are being tracked by Kubernetes.
+  It likely will mean we need to introduce the notion of tracked 
+  resources (which the operator watches but doesn't own) purely for the purpose of 
+  allowing references to those resources. 
+
+### Use built-in OwnerReference for owner references
+#### Pros
+- Basically none
+
+#### Cons
+- OwnerReference requires the object UID, which cannot be known at template authoring time.
+- OwnerReference only works for ownership relationships, not for 
```

Comment:
Were you going to finish with "references"? If so, that is intentional since they are for garbage collection, which a simple reference does not require.

Reply:
yes... fixed.